code/gui/serialbox.h

```diff
@@ -33,23 +36,27 @@ Q_OBJECT
 
     Q_SLOT void append_text(const std::string &text);
 
-private:
-    enum Power {
-        POWER_INTERVAL = 8,
-        POWER_MAX = 255,
-        POWER_MIN = 0
-    };
-
-    enum Delay {
-        DELAY_INTERVAL = 10,
-        DELAY_MAX = 150,
-        DELAY_MIN = 10,
-        DELAY_DEFAULT = 50
-    };
+    Q_SLOT void up_slider_changed(int value);
+    Q_SLOT void down_slider_changed(int value);
+    Q_SLOT void left_slider_changed(int value);
+    Q_SLOT void right_slider_changed(int value);
+
+    Q_SLOT void up_box_changed();
+    Q_SLOT void down_box_changed();
+    Q_SLOT void left_box_changed();
+    Q_SLOT void right_box_changed();
+
+    void slider_changed(int value, int dir);
 
+    void box_changed(int dir);
+
+private:
     Ui::SerialBox *ui;
     std::shared_ptr<Solenoid> m_solenoid;
 
+    QSlider *m_sliders[4];
+    QLineEdit *m_edit_boxes[4];
```

Comment:
If these aren't dynamically allocated can you replace them with weak_ref? You can do this in a different PR